src/main/java/com/uber/cadence/internal/worker/ActivityPollService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.internal.worker;
+
+import com.uber.cadence.*;
+import com.uber.cadence.internal.metrics.MetricsType;
+import com.uber.cadence.serviceclient.IWorkflowService;
+import com.uber.m3.tally.Stopwatch;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ActivityPollService
```

Comment:
PollService sounds like a service that performs polling, but as I understand this is class that polls the Cadence service.
I would rename this method to ActivityPollTask

Reply:
resolved